### DISCUSSION TITLE
Possible UI issues - ComboView not working  (Main UI in Part Design 1.0 RC4)

### DISCUSSION THREAD

**Discussion by lukolszewski** [FreeCAD/FreeCAD#17934] [Discussions that will turn in to issues but need refining]
I've been using FreeCAD from version 0.17 and I was quite excited to download 1.0 RC4. But I encountered issues from the very start. I started the app. I selected a new file. It  created a new file and brought me into the Part Design workbench.

Then I clicked (in tooltips) to create a body, then to create a sketch. And here my problem starts. Intead of the left panel having two tabs as before, one containing the Model and the other being an input for the current tool. We now have a Model tab and very narrow bit on the bottom that contains the input:
![image](https://github.com/user-attachments/assets/e1245ac2-8264-47d5-bd1c-fd9b652b4730)
OK, I thought one of these buttons on the top probably swithces to the previous view. So I tried the left most and I got this:
![image](https://github.com/user-attachments/assets/9812e167-5021-4f6f-b55e-26e36bfbc69d)
And when my mouse entered where the panel should be it would change to this:
![image](https://github.com/user-attachments/assets/05c0bb02-4478-456c-b297-25c91e14c218)

This view appears to be broken...

So I click the button again and now I have the panel with the tool entry stuff, good. But what is this on the right? The model window now became a "floating" panel, transparent normally, so not a big deal, but the moment the mouse enters its area the background switches to grey and it obscures half of the 3d view.
![image](https://github.com/user-attachments/assets/3c5695f7-81db-43a5-a251-1904946443a1)

I can close this panel, but then its gone... I don't see it once I finished editing the sketch. And how to switch back and forth between the sketch and the model?

Please let me know, Is this some graphics driver issue, or is it supposed to work this way? If it is, I think this is not a change for the better.

Also, why on earth would one change the constraints in Sketcher to now require TWO clicks instead of one?  What do I mean? Before I could click for a horizontal, or vertical or radius consrtaint as a single click. Now, I have to click "Dimension" and then select the type of consrtaint I want. I though, perhaps this is now a "smart" feature. Perhaps I just need to click the "Dimension" and it will know if I'm clicking a horizontal, a vertical line or a circle. Nope... I tried applying a horizontal constraint, then I clicked "Dimension" (without selecting vertical) and then I clicked on a vertical line thinknig that perhaps it will know... An error showed up same as if I just tried to apply a horizontal constraint to a vertical line. It simply remembered my last choice like previously radius/diameter tools did.

I can understand wanting to declutter the interface, but these are literallly the most clicked buttons there are. Hiding them behind an additional click is not a right choice. (Unless there is some way to bring back the old behaviour, if so, please let me know how?)

**Comment by hyarion** [comment]
I can agree that the panel buttons are a bit confusing and clicking the buttons randomly can lead to a situation where it isn't obvious how to recover from.

You managed to first change it into overlay and then hide them to the left. It isn't obvious that they are hidden (a small always showing ui indicator could be added to avoid this confusion)

To access these panels you either move the cursor to the left side of the 3d view and click on the ui item that pops up, or use the short cut [ctrl + left arrow key] (cmd + left arrow key on mac)

Regarding the constraints in sketcher, if you click the https://wiki.freecad.org/Sketcher_ConstrainHorVer button it will be smart and choose horizontal constraints for horizontal-ish lines and vertical constraint for vertical-ish lines (the threshold is at 45 deg). Only one click is required - and now with just one smart tool (or two clicks if you want to use a horizontal constraint on a vertical-ish line).

The smart dimension tool is for dimensions and the smart horizontal/vertical tool is for horizontal and vertical constraints. If you expect a tool to do one thing but it is designed to do something else, then yes indeed, it will not work as expected.
